### DISCUSSION TITLE
what's the use of address in contractQuery params

### DISCUSSION THREAD

**Discussion by dellwatson** [scio-labs/use-inkathon#75] [General]
As the title said,

`contractQuery(api, address, contract, method, options?, args?):`

the contract itself already has address, then what's the use of address here in this fnction again? 
if there's no address in the contracts list, it goes stale instead of using the address from contractQuery level

**Comment by wottpal** [comment]
It's the sender address, not the contract address. Also, it's optional and you can just pass an empty string (could be better typed, I agree). It's just to stay aligned with the polkadot.js `contract.query` function declaration ([see here](https://polkadot.js.org/docs/api-contract/start/contract.read#reading-contract-values)).

**Reply by dellwatson** [reply]
how to make it dynamic then? there's a scenario where i dont know the exact contract address yet. for e.g, contract address from backend or from input

**Reply by wottpal** [reply]
Just overwrite the `address` parameter in the [`useContract`](https://scio-labs.github.io/use-inkathon/functions/useContract.html) function!
